Sources/Commands/SwiftTestTool.swift

```diff
@@ -96,25 +121,44 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
         case .listTests:
             let testPath = try buildTestsIfNeeded(options)
             let testSuites = try getTestSuites(path: testPath)
+            let tests = testSuites.filteredTests(specifier: options.testCaseSpecifier)
+
             // Print the tests.
-            for testSuite in testSuites {
-                for testCase in testSuite.tests {
-                    for test in testCase.tests {
-                        print(testCase.name + "/" + test)
-                    }
-                }
+            for test in tests {
+                print(test.specifier)
             }
 
-        case .run(let specifier):
+        case .runSerial:
             let testPath = try buildTestsIfNeeded(options)
-            let success: Bool = TestRunner(path: testPath, xctestArg: specifier, processSet: processSet).test()
-            exit(success ? 0 : 1)
+            let testSuites = try getTestSuites(path: testPath)
+            var ranSuccessfully = true
+
+            if case .none = options.testCaseSpecifier {
```

Comment:
Sure, but having a switch with all case is always safer. If we modify the enum in future, the compiler will force us look at all the use sites, which is never bad. For e.g., if we add some case here and it should have been handled here, could go missed because it compiled successfully.
Does that make sense?